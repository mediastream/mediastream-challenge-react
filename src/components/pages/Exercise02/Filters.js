export default function Filters({
  loading,
  genres,
  order,
  handleChangeGenre,
  handleChangeOrder,
}) {
  return (
    <div className="movie-library__actions">
      <select
        name="genre"
        placeholder="Search by genre..."
        onChange={(event) => handleChangeGenre(event.target.value)}
        disabled={loading}
      >
        {genres.map((genre) => (
          <option value={genre} key={genre}>
            {genre}
          </option>
        ))}
      </select>
      <button onClick={() => handleChangeOrder()} disabled={loading}>
        Year {order === "asc" ? "Ascending" : "Descending"}
      </button>
    </div>
  );
}
