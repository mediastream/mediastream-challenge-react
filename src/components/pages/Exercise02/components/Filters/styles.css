.movie-library__actions {
  display: flex;
  flex-direction: row;
  margin-bottom: 1em;
}

.movie-library__actions .movie-library__actions__select {
  flex: 1;
  min-height: 45px;
  border-radius: 0.3em 0 0 0.3em;
  border: solid var(--ms-green) 3px;
  display: flex;
  margin-bottom: 0;
}

.movie-library__actions .movie-library__actions__select select {
  padding: 10px;
  width: 100%;
  border: 0;
  border-right: 10px solid white;
  font-size: 16px;
}

.movie-library__actions .movie-library__actions__button {
  border: solid var(--ms-green) 3px;
  border-radius: 0 0.3em 0.3em 0;
  white-space: nowrap;
  background-color: var(--ms-green);
  border: none;
  width: 200px;
  cursor: pointer;
  min-height: 45px;
  font-size: 16px;
}

@media (max-width: 425px) {
  .movie-library__actions .movie-library__actions__select,
  .movie-library__actions .movie-library__actions__button {
    min-height: 40px;
    font-size: 14px;
  }

  .movie-library__actions .movie-library__actions__button  {
    width: 140px;
  }
}

@media (max-width: 375px) {
  .movie-library__actions {
    flex-direction: column;
  }

  .movie-library__actions .movie-library__actions__select {
    margin-bottom: 8px;
    border-radius: 0.3em;
  }

  .movie-library__actions .movie-library__actions__button  {
    width: 100%;
    border-radius: 0.3em;
  }
}
