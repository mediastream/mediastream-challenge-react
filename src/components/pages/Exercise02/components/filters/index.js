const Filters = ({ loadingGenres, genres, genre, onChange, onClick, order, loading}) => {
    return (
        <div className="movie-library__actions">
            {(
                <select
                    disabled={loadingGenres || genres.length === 0}
                    name="genre"
                    placeholder="Search by genre..."
                    value={genre}
                    onChange={onChange}
                >
                    <option value="">
                        All genres
                    </option>
                    {genres.map(genre => (
                        <option key={genre} value={genre}>
                            {genre}
                        </option>
                    ))}
                </select>)}
            <button onClick={onClick} disabled={loading}>
                Year {order === 'desc' ? 'Descending' : 'Ascending'}
            </button>
        </div>
    )
}

export default Filters
