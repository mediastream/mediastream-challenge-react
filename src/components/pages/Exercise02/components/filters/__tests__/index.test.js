import { render } from '@testing-library/react'
import Filters from '..'

describe('Filter', () => {
    it('should render correctly', () => {
        const { container } = render(<Filters
            loadingGenres={false}
            genres={['Action','Sci-Fi']}
            genre={'Action'}
            onChange={() => {}}
            onClick={() => {}}
            loading={false}
            order={'desc'}
        />)
        expect(container).toMatchSnapshot()
    })
    it ('should render correctlyand click button', () => {
        const { container, getByText } = render(<Filters
            loadingGenres={false}
            genres={['Action','Sci-Fi']}
            genre={'Action'}
            onChange={() => {}}
            onClick={() => {}}
            loading={false}
            order={'asc'}
        />)
        expect(container).toMatchSnapshot()
        getByText('Year Ascending').click()
        expect(container).toMatchSnapshot()
    })
})