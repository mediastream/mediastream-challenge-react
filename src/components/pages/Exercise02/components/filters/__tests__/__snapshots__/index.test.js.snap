// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter should render correctly 1`] = `
<div>
  <div
    class="movie-library__actions"
  >
    <select
      name="genre"
      placeholder="Search by genre..."
    >
      <option
        value=""
      >
        All genres
      </option>
      <option
        value="Action"
      >
        Action
      </option>
      <option
        value="Sci-Fi"
      >
        Sci-Fi
      </option>
    </select>
    <button>
      Year 
      Descending
    </button>
  </div>
</div>
`;

exports[`Filter should render correctlyand click button 1`] = `
<div>
  <div
    class="movie-library__actions"
  >
    <select
      name="genre"
      placeholder="Search by genre..."
    >
      <option
        value=""
      >
        All genres
      </option>
      <option
        value="Action"
      >
        Action
      </option>
      <option
        value="Sci-Fi"
      >
        Sci-Fi
      </option>
    </select>
    <button>
      Year 
      Ascending
    </button>
  </div>
</div>
`;

exports[`Filter should render correctlyand click button 2`] = `
<div>
  <div
    class="movie-library__actions"
  >
    <select
      name="genre"
      placeholder="Search by genre..."
    >
      <option
        value=""
      >
        All genres
      </option>
      <option
        value="Action"
      >
        Action
      </option>
      <option
        value="Sci-Fi"
      >
        Sci-Fi
      </option>
    </select>
    <button>
      Year 
      Ascending
    </button>
  </div>
</div>
`;
